src/collector.rs

```diff
@@ -694,52 +705,170 @@ impl Collector {
     }
 }
 
-impl Recorder for RecorderHandle {
-    fn register_counter(&self, key: &Key, unit: Option<Unit>, description: Option<&'static str>) {
-        let _ = self.sender.try_send(Datum {
+pub struct RecorderHandle {
+    sender: mpsc::Sender<Datum>,
+}
+
+impl RecorderHandle {
+    pub fn register_counter(&self, key: &Key) -> metrics::Counter {
+        metrics::Counter::from_arc(Arc::new(RecorderHandleKey {
```

Comment:
This effectively creates temporary allocations in the typical usage like

```rust
metrics::counter!("test").increment(1);
```
which is going to be horrible for perfomance. We need to cache these handles to amortize the cost to make it berable. `metrics_util` has https://docs.rs/metrics-util/latest/metrics_util/registry/struct.Registry.html for this but I am not quite sure how we could use that while having time bucketed metrics to report to cloudwatch so either our own cache or maybe `metrics_util` could be extended https://github.com/metrics-rs/metrics/issues/546